src/components/content/exercise.tsx

```diff
@@ -31,33 +67,40 @@ export function Exercise(props: ExerciseProps) {
   } = props
   const [solutionVisible, setVisible] = React.useState(false)
 
-  let taskValue = task.children
-  let solutionValue = solution.children
+  let taskValue = {}
+  let solutionValue: unknown[] = []
   let interactiveComp = null
 
-  if (taskValue.length === 1 && taskValue[0].type === '@edtr-io/exercise') {
-    const state = taskValue[0].state
+  if (
+    task.children.length === 1 &&
+    task.children[0].type === '@edtr-io/exercise'
+  ) {
+    const state = task.children[0].state
     taskValue = state.content
     if (state.interactive) {
       if (state.interactive.plugin === 'scMcExercise') {
         interactiveComp = (
           <ScMcExercise
-            state={state.interactive.state}
+            state={state.interactive.state as ScMcExerciseProps['state']}
             idBase={`ex-${positionOnPage}-${positionInGroup}-`}
           />
         )
       }
       if (state.interactive.plugin === 'inputExercise') {
-        interactiveComp = <InputExercise data={state.interactive.state} />
+        interactiveComp = (
```

Comment:
Nitpicking:
- I'd wouldn't suffix `interactiveComp` with `Comp` since it is not a component (but a React Node)
- I'd usually move that logic into its own function, mostly because I don't like non-const variables that are global to the function but differently modified in the if-else clauses. By moving that into its own function, you can instead forego with the variable altogether.